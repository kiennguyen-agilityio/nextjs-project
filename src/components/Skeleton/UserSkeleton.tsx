const UserSkeleton = () => (
  <table className="w-full table-auto">
    <tbody>
      {[...Array(5)].map((_, index) => (
        <tr
          key={index}
          className="flex flex-col sm:flex-row items-start sm:items-center sm:gap-10 pr-8 md:pr-0 justify-between p-0 sm:p-4 border-b border-gray-200 bg-gray-100"
        >
          <td className="flex items-center">
            <div className="w-10 h-10 bg-gray-300 rounded-lg mr-4 animate-pulse" />
          </td>
          <td className="text-sm w-full truncate mb-4 sm:mb-0">
            <div className="h-4 bg-gray-300 rounded-md animate-pulse w-full" />
          </td>
          <td className="text-sm w-full truncate mb-4 sm:mb-0">
            <div className="h-4 bg-gray-300 rounded-md animate-pulse w-full" />
          </td>
          <td className="text-sm w-full truncate mb-4 sm:mb-0">
            <div className="h-4 bg-gray-300 rounded-md animate-pulse w-full" />
          </td>
          <td className="w-full sm:w-1/12">
            <div className="w-6 h-6 bg-gray-300 rounded-full animate-pulse" />
          </td>
        </tr>
      ))}
    </tbody>
  </table>
);

export default UserSkeleton;
