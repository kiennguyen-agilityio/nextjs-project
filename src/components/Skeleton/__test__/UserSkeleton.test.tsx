import { render } from '@testing-library/react';

// component
import UserSkeleton from '@/components/Skeleton/UserSkeleton';

describe('RoleSkeleton Component', () => {
  it('renders without crashing and matches snapshot', () => {
    const { asFragment } = render(<UserSkeleton />);
    expect(asFragment()).toMatchSnapshot();
  });
});
