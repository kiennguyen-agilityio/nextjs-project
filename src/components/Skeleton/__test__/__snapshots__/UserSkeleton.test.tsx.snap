// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoleSkeleton Component renders without crashing and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="w-10 h-10 bg-gray-300 rounded-lg mr-4 animate-pulse"
  />
  <div
    class="w-1/4 h-4 bg-gray-300 rounded-md animate-pulse"
  />
  <div
    class="w-6 h-6 bg-gray-300 rounded-full animate-pulse"
  />
</DocumentFragment>
`;
